### ISSUE TITLE
restrict run command to a substate

### ISSUE THREAD

**Issue by ghost** [SoarGroup/Soar#52]
_From [joseph...@gmail.com](https://code.google.com/u/113587566993378695814/) on August 05, 2009 10:43:29_

Recently I've been doing some experiments where it was useful to be
able to run the agent until a goal retracts. I implemented this as an
option to the run command, -g or --goal. I think this functionality is
useful for other people as well, so I'm submitting this as an
enhancement request. The actual work is already done.

Basically, I added a flag on the Agent struct called
stop_on_substate_removal. When this flag is set and
remove_existing_context_and_descendents is called,
thisAgent->stop_soar will be incremented, and the flag reset, so that
Soar will be interrupted at the next opportunity.

_Original issue: http://code.google.com/p/soar/issues/detail?id=51_
